src/common/message-creators/assessment-card-selection-message-creator.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import {
+    BaseActionPayload,
+    AssessmentCardSelectionPayload,
+    RuleExpandCollapsePayload,
+} from 'background/actions/action-payloads';
+import { TelemetryEventSource } from 'common/extension-telemetry-events';
+import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
+import { ActionMessageDispatcher } from 'common/message-creators/types/dispatcher';
+import { Messages } from 'common/messages';
+import { SupportedMouseEvent, TelemetryDataFactory } from 'common/telemetry-data-factory';
+
+export class AssessmentCardSelectionMessageCreator implements CardSelectionMessageCreator {
+    constructor(
+        private readonly dispatcher: ActionMessageDispatcher,
+        private readonly telemetryFactory: TelemetryDataFactory,
+        private readonly source: TelemetryEventSource,
+    ) {}
+
+    public toggleCardSelection = (
+        ruleId: string,
+        resultInstanceUid: string,
+        event: React.SyntheticEvent,
+        testKey: string,
+    ) => {
+        const payload: AssessmentCardSelectionPayload = {
+            testKey,
+            resultInstanceUid,
+            ruleId,
+            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
+        };
+
+        this.dispatcher.dispatchMessage({
+            messageType: Messages.AssessmentCardSelection.CardSelectionToggled,
+            payload,
+        });
+    };
+
+    public toggleRuleExpandCollapse = (ruleId: string, event: React.SyntheticEvent) => {
+        const payload: RuleExpandCollapsePayload = {
+            ruleId,
+            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
+        };
+
+        this.dispatcher.dispatchMessage({
+            messageType: Messages.AssessmentCardSelection.RuleExpansionToggled,
+            payload,
+        });
+    };
+
+    public collapseAllRules = (event: SupportedMouseEvent) => {
+        const payload: BaseActionPayload = {
```

Comment:
These are still all the old payloads, we want to make sure that these payload types match the type associated with the message we're sending and include the `TestKey`. For example, this `collapseAllRules` message eventually triggers the `collapseAllRules` callback in the `AssessmentCardSelectionStore`, which accepts a `AssessmentExpandCollapsePayload`. With this current implementation we're sending it a `BaseActionPayload` instead, so we want to change line 53 to be `const payload: AssessmentExpandCollapsePayload = {`.

This will also result in a type checking error that we need to include the testKey, which we can add as a parameter to this method similarly to how you did in the `toggleCardSelection` method on line 21. Also note that this change will need to be reflected in the `CardSelectionMessageCreator` interface- so all of those interface methods will have an additional optional test key parameter after this change is finished. 

The same goes for the other payloads in this file (lines 41, 64, and 75), we need to make sure their types match what the store callbacks expect.